File: java - oop/src/main/java/Ad.java
public class Ad extends BaseAdvertising {
    private String title;
    private String imgURL;
    private String link;
    private Advertiser advertiser;

    public Ad(int id, String title, String imgURL, String link, Advertiser advertiser) {
        this.id = id;
        this.title = title;
        this.imgURL = imgURL;
        this.link = link;
        this.advertiser = advertiser;
        this.clicks_count = 0;
        this.views_count = 0;
    }

    @Override
    public String describeMe() {
        return "This is 'Ad class'. This class provides data about Ads such as title , imgURL , link ,clicks_count , views_count , advertiser and also can set and get this fields.";
    }

    @Override
    protected void incClicks() {
        super.incClicks();
        advertiser.incClicks();
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getImgURL() {
        return imgURL;
    }

    public void setImgURL(String imgURL) {
        this.imgURL = imgURL;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }
}

File: python - oop/pythonProject1/Ad.py
from BaseAdvertising import BaseAdvertising
from Advertiser import Advertiser


class Ad(BaseAdvertising):
    def __init__(self, id: int, title: str, imgURL: str, link: str, advertiser: Advertiser):
        super().__init__()
        self.id = id
        self.title = title
        self.imgURL = imgURL
        self.link = link
        self.advertiser = advertiser

    def describeMe(self) -> str:
        print(
            "This is 'Ad class'. This class provides data about Ads such as title , imgURL , link , advertiser and also can set and get this fields.")

    def incClicks(self):
        super().incClicks()
        self.advertiser.incClicks()

    def getTitle(self) -> str:
        return self.title

    def setTitle(self, title: str) -> str:
        self.title = title

    def getImgURL(self) -> str:
        return self.imgURL

    def setImgURL(self, imgURL: str) -> str:
        self.imgURL = imgURL

    def getLink(self) -> str:
        return self.link

    def setLink(self, link: str) -> str:
        self.link = link
